### ISSUE TITLE
[promise-function-async] False positive on async function with async keyword already existing

### ISSUE THREAD

**Issue by WickedSik** [typescript-eslint/typescript-eslint#1092]
**What code were you trying to parse?**

```ts
import fetch from 'cross-fetch'

export default class Client {
    constructor(private readonly uri: string) {
    }

    public async get(path: string): Promise<any> {
        return fetch(`http://${this.uri}${path}`)
            .then(response => response.json())
    }

    public async post(path: string, body: string | FormData | URLSearchParams): Promise<any> {
        return fetch(`http://${this.uri}${path}`, {
            method: 'POST',
            body
        }).then(response => response.json())
    }
}
```

**What did you expect to happen?**
No errors concerning missing `async` keywords

**What actually happened?**
```
dev@PC:~/project$ npx eslint --fix src/api/Client.ts 

/home/dev/project/src/api/Client.ts
   9:19  error  Functions that return promises must be async  @typescript-eslint/promise-function-async
  16:17  error  Functions that return promises must be async  @typescript-eslint/promise-function-async

✖ 2 problems (2 errors, 0 warnings)
```

**Versions**

| package                                   | version |
| ----------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin-tslint` | `2.2.0` |
| `@typescript-eslint/parser`               | `2.2.0` |
| `TypeScript`                              | `3.6.2` |
| `ESLint`                                  | `6.4.0` |
| `node`                                    | `8.16.1` |
| `npm`                                     | `6.12.0` |

**Comment by bradzacher** [comment]
I am unable to repro this against master.
Looking at both the AST of the code and the code of the rule, this should not occur.

**Comment by WickedSik** [comment]
I will check first with the latest public version, then with master. I haven't ran `npm update` in a few weeks.

**Comment by WickedSik** [comment]
@bradzacher Running with both `2.4.0` and `canary` gives the same result as above.

This is including the update of `eslint` to `6.5.1`

**Comment by amikheychik** [comment]
Can confirm the issue, though it's intermittent. Out of three functions in the same file (and who knows how many in the whole project), only one fails this check.

**Comment by buu700** [comment]
Is this a false positive? I would expect both instances of `response => response.json()` to have to be `async response => response.json()` since `response.json` returns a promise.

(I wrote the original equivalent rule in tslint.)

**Comment by WickedSik** [comment]
@buu700 @bradzacher This is actually the solution. When adding `async` to `response => response.json()` the error disappears.

This issue can be closed as it's not actually a bug or false positive.
